src/CommunicationPattern/CommunicationPattern.H

```diff
@@ -0,0 +1,189 @@
+// SPDX-FileCopyrightText: 2024 OGL authors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <ginkgo/ginkgo.hpp>
+
+#include "fvCFD.H"
+
+#include "DevicePersistent/ExecutorHandler/ExecutorHandler.H"
+
+using CommCounts = std::tuple<std::vector<label>, std::vector<label>,
+                              std::vector<label>, std::vector<label>>;
+
+/* @brief  This function computes the send and recv counts vectors for
+ * for scattering from an owner to non  owner ranks
+ *
+ * @param exec_handler The executor handler
+ * @param ranks_per_owner Ratio of total ranks to owner ranks
+ * @param size how many elements to send to owner
+ * */
+CommCounts comput_scatter_from_owner_counts(const ExecutorHandler &exec_handler,
+                                            label ranks_per_owner, label size);
+
+/* @brief  This function computes the send and recv counts and offset vectors
+ *for repartitioning to an owner rank
+ *
+ * @param exec_handler The executor handler
+ * @param ranks_per_owner Ratio of total ranks to owner ranks
+ * @param size how many elements to send to owner
+ * @param total_size Size of the total buffer
+ * @param padding_before Number of unused elements before
+ * @param padding_after Number of unused elements after
+ * @return The CommCounts
+ * */
+CommCounts compute_gather_to_owner_counts(const ExecutorHandler &exec_handler,
+                                          label ranks_per_owner, label size,
+                                          label total_size,
+                                          label padding_before,
+                                          label padding_after);
+
+/* @brief  This function computes the send and recv counts and offset vectors
+ *for repartitioning to an owner rank
+ *
+ * @param exec_handler The executor handler
+ * @param ranks_per_owner Ratio of total ranks to owner ranks
+ * @parameter size how many elements to send to owner
+ * @return The CommCounts
+ * */
+CommCounts compute_gather_to_owner_counts(const ExecutorHandler &exec_handler,
+                                          label ranks_per_owner, label size);
+
+/* @brief  This function computes the send and recv counts vectors for
+ * for communication back to neighbours from an owner rank
+ *
+ * @param exec_handler The executor handler
+ * @param ranks_per_owner Ratio of total ranks to owner ranks
+ * @parameter size how many elements to send to owner
+ * @return The CommCounts
+ * */
+CommCounts compute_scatter_from_owner_comm(const ExecutorHandler &exec_handler,
+                                           label ranks_per_owner, label size);
+
```

Comment:
This purpose of this function is similar to the purpose of another function `comput_scatter_from_owner_counts`
Why do we need these two similar functions?

Reply:
Good catch, this function isn't needed.